### ISSUE TITLE
[Feature Request]: Integration of Rocket.Chat

### ISSUE THREAD

**Issue by stbc** [grokability/snipe-it#11123]
### Is your feature request related to a problem? Please describe.

First things first: Congrats for v6!
I read, that in version 6 there is a 

> stricter validation for slack hooks (people were trying to use non-slack urls for webhooks)

which is exactly what we do, as we are using a rocket.chat hook on our self hosted rocket.chat server. It worked like a charm, right out of the box. No tweaking necessary.

### Describe the solution you'd like

Integrate the rocket.chat hook as an official webhook into snipe-it. As the Slack integration also works with RC, it should be only a little tweak.

**Comment by snipe** [comment]
I'm not familiar with RC - what exactly would need to be done here?

**Comment by stbc** [comment]
Allow rocket-URLs. 
Works with exactly the same payload in the json response. But there is no central Rocket-Host or something, as the most servers are self hosted. 
![grafik](https://user-images.githubusercontent.com/2520567/168811826-d0228234-8181-4017-89e9-f2bc3275aa5b.png)
Maybe a second block für non-slack-urls would work. Before the check was introduced, it worked with exactly the same code
![grafik](https://user-images.githubusercontent.com/2520567/168812622-7520a30a-b6be-45e9-b9af-aa9cf777c0f3.png)

**Comment by woloss** [comment]
> Allow rocket-URLs. Works with exactly the same payload in the json response. But there is no central Rocket-Host or something, as the most servers are self hosted. 

You can edit validation manually.  Just swap ```https://hooks.slack.com``` to ```https://```
If using docker use this paths ```vi var/www/html/app/Http/Requests/SlackSettingsRequest.php``` and probably ```vi var/www/html/app/Http/Controllers/Api/SettingsController.php```

However, messages are empty. Need to look further. (Test button uses another payload)

UPD: Due to this issue https://github.com/RocketChat/Rocket.Chat/issues/25825. Can't really debug custom script.

UPD2: Turned out snipe-it have other payload
<details><summary>Custom scriptfor "Test # Integration" button</summary>

```
class Script {
  process_incoming_request({ request }) {
    console.log(request.content_raw)
    var result = JSON.parse(request.content_raw);

    return {
      content:{
        text: result.text,
      }
    };
  }
}
```
</details>

As for proper notifications(checkouts, etc) - https://docs.rocket.chat/guides/administration/admin-panel/integrations

**Comment by valireality** [comment]
Just change `https://hooks.slack.com` with your RC URL in `app/Http/Requests/SlackSettingsRequest.php` and `app/Http/Controllers/Api/SettingsController.php` and it will work :)

I confirm that this hack is working with Rocket.Chat.

**Comment by JoKneeMo** [comment]
I can confirm this works for Mattermost self hosted as well!
Can the url validation be removed so that we don't have to manually modify this after updates?
I can submit a PR if required.

**Comment by snipe** [comment]
We'd love to, but the problem is that bad slack (or slack adjacent) urls can sometimes cause 500 errors, since the server can't be reached or the API token is wrong, or the webhook defined is no longer valid. It was shitting up our logs, and causing real frustration with our customers and users, because they didn't know the slack endpoint was donked. 

I'm certainly happy to extend the validation to include additional services (like rocketchat and mattermost) if you can give me a url pattern that we can add to validation. 

At the end of the day, what you're experiencing is the difference between n00bs and power users, and we had to nerf this one. I have zero issues adding in some additional url schemes though.

**Comment by JoKneeMo** [comment]
Thanks for the detailed reasoning, but I'm afraid I would not be able to provide any additional urls since all of my installs of Rocket.Chat and Mattermost are on-premise.

I was using it fine with Rocket.Chat in v5.1 but v6 prevented the endpoint from changing.

I'm now using Mattermost and have migrated Snipe over to docker.

I've put the below snippet in the yaml for the main snipe-it service:

```yaml
command:
  - /bin/bash
  - -c
  - |
    sed -i 's|https://hooks.slack.com|https://|' app/Http/Requests/SlackSettingsRequest.php
    sed -i 's|https://hooks.slack.com/|https://|' app/Http/Controllers/Api/SettingsController.php
    /startup.sh
```
It's just changing the validator to make sure the endpoint is using TLS.

For the record, the Slack webhook syntax was ported over to most every chat service for incoming webhooks.

**Comment by snipe** [comment]
Yes, but we still need to validate the URLs somehow. You would be amazed at the URLs people were putting in there before, and then they'd hit 500s on checkout. I want to be flexible here, but I also need to make sure we're validating properly.

**Comment by JoKneeMo** [comment]
This issue can be closed as it was addressed in #12570 and released in v6.1.0.
